File: src/front/web-client/src/app/models/constituency.model.ts
import { LocationLevel } from '@app/services/nswag/api-nswag-client';

export interface ConstituencyNode {
  id: number;
  code: string;
  wording: string;
  parentId?: number;
  level: LocationLevel;
  children?: ConstituencyNode[];
  expanded?: boolean;
}

File: src/back/src/Backend/11-Application.Features/Common/Constituency/ConstituencyModel.cs
using Application.Common.Enums;

namespace Application.Features.Common.Constituency
{
    public class ConstituencyModel
    {
        public string? Code { get; set; }
        public string? Wording { get; set; }
        public LocationLevel Level { get; set; }
        public long? ParentId { get; set; }
        public bool IsActive { get; set; }
    }
}
